### PR TITLE
343: Create Skeleton Records Validation errors

### PR DESCRIPTION
**Pull request by ChazUK**
343: Fixing issue where it was not calculating amount of staff records added
343: Adding Validation for staff records to be required

### CODE REVIEW

**Comment by ChazUK** [comment]
[Add skeleton records: Not taking into account the number of workers added](https://trello.com/c/Nmc4R1yR/343-add-skeleton-records-not-taking-into-account-the-number-of-workers-added)

File: src/app/features/workers/create-staff-record/create-staff-record.component.ts [hunk 1]
```diff
@@ -150,6 +157,18 @@ export class CreateStaffRecordComponent implements OnInit, OnDestroy {
     return null;
   }
 
+  staffRecordsRequiredValidator() {
+    if (this.form) {
+      const staffRecordsControl = <FormArray>this.form.controls.staffRecords;
```
Comment (denny13) [inline src/app/features/workers/create-staff-record/create-staff-record.component.ts]: The same const is used in 3 places could be more of a public variable, or are they required to each be private?

Not necessary to change as im sure it works fine, just a check.

File: src/app/features/workers/create-staff-record/create-staff-record.component.ts [hunk 2]
```diff
@@ -46,26 +47,30 @@ export class CreateStaffRecordComponent implements OnInit, OnDestroy {
   ngOnInit() {
     this.subscriptions.add(
       this.establishmentService.getStaff().subscribe(establishmentStaff => {
-        this.establishmentStaff = establishmentStaff;
-        this.form.controls.totalStaff.patchValue(establishmentStaff);
-        this.form.updateValueAndValidity();
+        if (establishmentStaff) {
+          this.establishmentStaff = establishmentStaff;
```
[the remark this answers is not in the document]
Comment (ChazUK) [reply src/app/features/workers/create-staff-record/create-staff-record.component.ts]: Removed.

File: src/app/features/workers/create-staff-record/create-staff-record.component.ts (hunk 1 above)
Comment (ChazUK) [reply src/app/features/workers/create-staff-record/create-staff-record.component.ts]: Updated this to use a get function

**Comment by ChazUK** [comment]
[Add skeleton records: Error message on entry to the page](https://trello.com/c/AFsuw9gF/345-add-skeleton-records-error-message-on-entry-to-the-page)

**Pull request merged**